src/ess/isissans/mantidio.py

```diff
@@ -99,12 +100,29 @@ def from_data_workspace(
 
 
 def load_run(filename: Filename[RunType], period: Period) -> DataWorkspace[RunType]:
-    loaded = _mantid_simpleapi.Load(
-        Filename=str(filename), LoadMonitors=True, StoreInADS=False
-    )
+    # Loading many small files with Mantid is, for some reason, very slow when using
+    # the default number of threads in the Dask threaded scheduler (1 thread worked
+    # best, 2 is a bit slower but still fast). We can either limit that thread count,
+    # or add a lock here, which is more specific.
```

Comment:
Interesting. This might also help with other loaders. I saw some slowdown in BIFROST when using a threaded scheduler.

Reply:
You think this is due to HDF5?